### PR TITLE
Taking care of multiple line messages, closes #41

### LINKED ISSUE #41: Message sending fails on '\n'
If a message is sent with using the  `Reply` function, and the `text` argument contains a line feed character, only the part of text before that return is sent.

Example:
If  the following code is used, `bot.Reply(message, "AA\nBB\nCC")`

Only `AA` will be sent back.

While this usefully limits the output of a trigger, it can be annoying, especially when trying to format information.

If PRs are welcome, could I write a new `ReplyWithReturn` type function, different from the `Reply` function to avoid breaking backward compatibility ?

Thanks in advance.

### PR DESCRIPTION
**Pull request by m-242**
See [related issue](https://github.com/whyrusleeping/hellabot/issues/41) for more precisions.

### CODE REVIEW

File: hellabot.go [hunk 1]
```diff
@@ -270,11 +270,19 @@ func (bot *Bot) Reply(m *Message, text string) {
 
 // Msg sends a message to 'who' (user or channel)
 func (bot *Bot) Msg(who, text string) {
-	for len(text) > 400 {
-		bot.Send("PRIVMSG " + who + " :" + text[:400])
-		text = text[400:]
+	for _, line := range strings.Split(text, "\n") {
```
Comment (icholy) [inline hellabot.go:273]: This won't work for windows style newlines. Use a `bufio.Scanner`.

Comment (m-242) [reply hellabot.go]: Thanks for the tip ^^

File: hellabot.go [hunk 2]
```diff
@@ -270,11 +270,20 @@ func (bot *Bot) Reply(m *Message, text string) {
...
 // Msg sends a message to 'who' (user or channel)
 func (bot *Bot) Msg(who, text string) {
-	for len(text) > 400 {
-		bot.Send("PRIVMSG " + who + " :" + text[:400])
-		text = text[400:]
+	scanner := bufio.NewScanner(strings.NewReader(text))
+	for scanner.Scan() {
+		bot.sendline(who, scanner.Text())
+	}
+}
```
Comment (icholy) [inline hellabot.go:277]: Pull the splitting logic into a separate function which handles both newlines and long lines. Then re-use that function in the `Notice` call.

``` go
func splitText(s string) []string
```

Comment (m-242) [reply hellabot.go:276]: That function is used multiple times, shouldn't it be placed into a separate `utils.go` file ?

Comment (icholy) [reply hellabot.go:276]: Doesn't matter, I wouldn't bother.

Comment (m-242) [reply hellabot.go:276]: Thanks.

File: hellabot.go [hunk 3]
```diff
@@ -270,20 +270,38 @@ func (bot *Bot) Reply(m *Message, text string) {
...
+			for len(tmptxt) > 400 {
+				ret = append(ret, tmptxt[:400])
+				tmptxt = tmptxt[400:]
+			}
+			ret = append(ret, tmptxt)
+		} else {
+			ret = append(ret, scanner.Text())
+		}
+	}
+	return ret
```
Comment (icholy) [inline hellabot.go:297]:
``` go

// Splits a given string into a string slice, in chunks ending
// either with \n, or with \r\n, or of a size of 400 characters.
func splitText(text string) []string {
    var ret []string
    scanner := bufio.NewScanner(strings.NewReader(text))
    for scanner.Scan() {
	line := scanner.Text()
	for len(line) > 400 {
	    ret = append(ret, line[:400])
	    line = line[400:]
	}
	ret = append(ret, line)
    }
    return ret
}

```

Comment (m-242) [reply hellabot.go:298]: It's indeed a much cleaner and less complex version.

**Comment by icholy** [comment]
@m-242 LGTM. I'm going to hold off on merging until we talk about it more in the issue.

**Comment by m-242** [comment]
Of course, thanks for your interest and reviews.

**Comment by icholy** [comment]
@m-242 can you rebase/squash these commits?

**Comment by icholy** [comment]
@m-242 you had it right the first time. Looks like you botched the rebase the second time around.

**Comment by m-242** [comment]
Ok, I reset to `4433f5d` and forced pushed.

**Comment by icholy** [comment]
@m-242 thanks!

**Pull request merged**